zkevm-circuits/src/exp_circuit.rs

```diff
@@ -264,11 +273,12 @@ impl<F: Field> SubCircuitConfig<F> for ExpCircuitConfig<F> {
 
             cb.gate(and::expr([
                 meta.query_selector(q_usable),
-                meta.query_advice(exp_table.is_step, Rotation::cur()),
+                meta.query_fixed(exp_table.is_step, Rotation::cur()),
             ]))
         });
 
         Self {
+            minimum_rows: meta.minimum_rows(),
```

Comment:
I'm not sure if you need this field. You already know the minimum number of rows that this circuit will require (based on the number of steps in all `exp_events`. You also know the maximum number of steps that the circuit params allow. So use those 2 parameters to figure out at the time of witness assignment if the number of rows are sufficient or not.